nikola/plugins/command/status.py

```diff
@@ -0,0 +1,106 @@
+
+        try:
+            with io.open(timestamp_path, "r", encoding="utf8") as inf:
+                last_deploy = datetime.strptime(inf.read().strip(), "%Y-%m-%dT%H:%M:%S.%f")
+                last_deploy_offset = datetime.utcnow() - last_deploy
+        except (IOError, Exception):
+            print("It does not seem like you’ve ever deployed the site (or cache missing).")
+
+        if last_deploy:
+
+            if last_deploy_offset.days > 0:
+                last_deploy_offsetstr = "{0} days and {1} hours".format(str(int(last_deploy_offset.days)), str(int(last_deploy_offset.seconds / 60 / 60)))
+            elif last_deploy_offset.seconds / 60 / 60 > 0:
+                last_deploy_offsetstr = "{0} hours and {1} minutes".format(str(int(last_deploy_offset.seconds / 60 / 60)), str(int(last_deploy_offset.seconds / 60 - ((last_deploy_offset.seconds / 60) // 60) * 60)))
+            else:
+                last_deploy_offsetstr = "{0} minutes".format(str(int(last_deploy_offset.seconds / 60 - ((last_deploy_offset.seconds / 60) // 60) * 60)))
+
+            fmod_since_deployment = 0
+            for root, dirs, files in os.walk(self.site.config["OUTPUT_FOLDER"], followlinks=True):
+                if not dirs and not files:
+                    continue
+                for fname in files:
+                    fpath = os.path.join(root, fname)
+                    fmodtime = datetime.fromtimestamp(os.stat(fpath).st_mtime)
+                    if fmodtime.replace(tzinfo=tzlocal()) > last_deploy.replace(tzinfo=gettz("UTC")).astimezone(tz=tzlocal()):
+                        fmod_since_deployment = fmod_since_deployment + 1
+
+            if fmod_since_deployment > 0:
+                print("{0} output files modified since last deployment {1} ago.".format(str(fmod_since_deployment), last_deploy_offsetstr))
+            else:
+                print("Last deployment {0} ago.".format(last_deploy_offsetstr))
+
+        posts_count = len(self.site.all_posts)
+        posts_drafts = 0
+        posts_scheduled = 0
+        post_scheduled_nearest_offset = None
+
+        for post in self.site.all_posts:
+            if post.is_draft:
+                posts_drafts = posts_drafts + 1
+            if post.publish_later:
+                posts_scheduled = posts_scheduled + 1
+                post_due_offset = post.date - datetime.utcnow().replace(tzinfo=gettz("UTC"))
+                if (post_scheduled_nearest_offset is None) or (post_due_offset.seconds < post_scheduled_nearest_offset.seconds):
+                    post_scheduled_nearest_offset = post_due_offset
+
+        if posts_scheduled > 0 and post_scheduled_nearest_offset is not None:
+            if post_scheduled_nearest_offset.days > 0:
+                nearest_scheduled_timestr = "{0} days and {1} hours".format(str(int(post_scheduled_nearest_offset.days)), str(int(post_scheduled_nearest_offset.seconds / 60 / 60)))
```

Comment:
Deduplication: write a `display_time` function that takes `offset` as an argument and is used for generating pretty `last_deploy_offset` and `post_scheduled_nearest_offset` text